### ISSUE TITLE
static object in namespace reported as global variables

### ISSUE THREAD

**Issue by jwuttke** [CoatiSoftware/SourcetrailBugTracker#566]
namespace {
        static int foo = 4;
    }

results in foo being reported in the category "Global Variables".

cppreference.com says about the static specifier: "When used in a declaration at namespace scope, it specifies internal linkage."

**Comment by egraether** [comment]
Right, so it is not global because it is separately allocated in each source file.
What should we call it then?

**Comment by mlangkabel** [comment]
@jwuttke: Thank you for bringing this up. As you already quoted, it is not straight forward. I think we already addressed this at issue #514. The solution we came up with is bundling it as a global variable but appending the name of the translation unit, so that you get one symbol in Sourcetrail per translation unit using the static object at namespace scope.

For example if you would put the code from above in a header called `Foo.h` and include this header from inside 2 different source files `Foo1.cpp` and `Foo2.cpp`, then Sourcetrail would show 2 different symbols called `foo (Foo1.cpp)` and `foo (Foo2.cpp)`, each having its own set of edges pointing to it.

So from Sourcetrail's perspective these are global variables but they are not just merged to one single symbol. Instead Sourcetrail keeps track of the respective translation unit they belong to. Do you think this is enough? Or - maybe I should have asked this question first - is this mechanism working properly for your code?
